src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -2075,12 +2088,6 @@ SetForceEarlyReturn::doit(Thread *target, bool self) {
   if (java_thread->is_exiting()) {
     return; /* JVMTI_ERROR_THREAD_NOT_ALIVE (default) */
   }
-  if (!self) {
```

Comment:
Can't we have any racing by removing this check?
We are checking thread state before handshake operation, but it is changed before thread start execution of this handshake?

Reply:
Thank you for the comment.
No, there can be no race here. If a JVMTI function is called (or not called) on the current thread (eg. the target thread is current) then it can't change while the JVMTI function is executed.